src/Native/Unix/System.Native/pal_io.c

```diff
@@ -233,14 +248,20 @@ static int32_t ConvertOpenFlags(int32_t flags)
             return -1;
     }
 
+#if defined(_AIX)
+    if (flags & ~(PAL_O_ACCESS_MODE_MASK | PAL_O_CREAT | PAL_O_EXCL | PAL_O_TRUNC | PAL_O_SYNC))
+#else
     if (flags & ~(PAL_O_ACCESS_MODE_MASK | PAL_O_CLOEXEC | PAL_O_CREAT | PAL_O_EXCL | PAL_O_TRUNC | PAL_O_SYNC))
+#endif
```

Comment:
Why do we need this ifdef?  PAL_O_CLOEXEC is defined regardless of the native dependencies.